File: 2022/file/file.go
package file

import (
	"io/ioutil"
	"strings"
)

func ReadLines(file string) []string {
	data, err := ioutil.ReadFile(file)
	if err != nil {
		panic(err)
	}
	lines := strings.Split(string(data), "\n")

	return lines
}

File: 2024/common/file.ts
import fs from 'node:fs/promises';

export async function readLines(filename: string) {
  let file = '';
  try {
    file = await fs.readFile(filename, { encoding: 'utf-8' });
  } catch (e) {
    console.log(`Error reading from input file ${e}`);
    throw e;
  }
  const lines = file.split('\n');
  lines.pop();
  return lines;
}
